opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/AsyncBasicDetectorMinaImpl.java

```diff
@@ -206,26 +200,22 @@ private static final SSLContext createClientSSLContext() throws NoSuchAlgorithmE
      * @return IoFutureListener<ConnectFuture>
      */
     private final IoFutureListener<ConnectFuture> retryAttemptListener(final DetectFutureMinaImpl detectFuture, final InetSocketAddress address, final IoSessionInitializer<ConnectFuture> init, final int retryAttempt) {
-        return new IoFutureListener<ConnectFuture>() {
-
-            @Override
-            public void operationComplete(ConnectFuture future) {
-                final Throwable cause = future.getException();
+        return future -> {
+            final Throwable cause = future.getException();
 
-                if (cause != null) {
-                    if (cause instanceof IOException) {
-                        if (retryAttempt == 0) {
-                            LOG.info("Service {} detected false: {}: {}",getServiceName(), cause.getClass().getName(), cause.getMessage());
-                            detectFuture.setServiceDetected(false);
-                        } else {
-                            LOG.info("Connection exception occurred: {} for service {}, retrying attempt {}", cause, getServiceName(), retryAttempt);
-                            future = m_connectionFactory.reConnect(address, init, createDetectorHandler(detectFuture));
-                            future.addListener(retryAttemptListener(detectFuture, address, init, retryAttempt - 1));
-                        }
-                    } else {
-                        LOG.info("Threw a Throwable and detection is false for service {}", getServiceName(), cause);
+            if (cause != null) {
+                if (cause instanceof IOException) {
+                    if (retryAttempt == 0) {
+                        LOG.info("Service {} detected false: {}: {}",getServiceName(), cause.getClass().getName(), cause.getMessage());
```

Comment:
3 x {}, 4 arguments?

Reply:
...4?  I see:
1. `getServiceName()`
2. `cause.getClass().getName()`
3. `cause.getMessage()`

Reply:
OMG. I‘m sorry. Forget that.